src/libmongoc/tests/test-mongoc-crud.c

```diff
@@ -51,8 +51,135 @@ test_all_spec_tests (TestSuite *suite)
       suite, resolved, &test_crud_cb, TestSuite_CheckLive);
 }
 
+static void
+prose_test_1 (void *ctx)
```

Comment:
FWIW, I actually wrote this _before_ I realized the prose test had already been implemented in https://github.com/mongodb/mongo-c-driver/commit/84635b39d4a3cdde133c3c8cfebd9c070a1943ab. Aside from some stylistic differences, it looks exactly the same. Seeing the original code did motivate me to use `get_test_collection` in both of these tests.

Let me know if you prefer to keep the original function, but either way I think we should relocate it to this file. That's consistent with how we organize prose tests for other specs (e.g. change streams, transactions).

Reply:
Relocating the test seems best to me (and this new version is nicely commented + formatted). I see prose tests for change streams, transactions and SDAM are in the right places; are there other prose tests that are scattered amongst unit tests?

Reply:
I've no idea if we have other prose tests scattered about. I only found this original prose test because I happened to check the commit for the DRIVERS ticket for when this test was introduced. Nothing about the name of the test I deleted suggested that it was a spec prose test.

Reply:
Moving and renaming to be more consistent SGTM. That should make it easier for readers to tie it back to the spec prose test.